bookwyrm/templates/feed/feed.html

```diff
@@ -16,10 +16,45 @@ <h1 class="title">
     </ul>
 </div>
 
+{# feed settings #}
+<details class="mb-5" {% if settings_saved %}open{% endif %}>
+    <summary>
+        <span class="has-text-weight-bold">
+            {{ _("Feed settings") }}
+        </span>
+        {% if settings_saved %}
+            <span class="tag is-success is-light ml-2">{{ _("Saved!") }}</span>
+        {% endif %}
+    </summary>
+    <form class="level is-align-items-flex-end" method="post" action="/{{ tab.key }}#feed">
+        {% csrf_token %}
+
+        <div class="level-left">
+            <div class="field">
+                <div class="control">
+                    <label class="label mt-2 mb-1">Status types</label>
+                    {% for name, value in feed_status_types_options %}
+                    <label class="mr-2">
+                        <input type="checkbox" name="feed_status_types" value="{{ name }}" {% if name in user.feed_status_types %}checked=""{% endif %}/>
+                        {{ value }}
+                    </label>
+                    {% endfor %}
+                </div>
+            </div>
+        </div>
+        <div class="level-right control">
+            <button class="button is-small is-primary is-outlined" type="submit">
+                {{ _("Save settings") }}
+            </button>
+        </div>
+    </form>
+</details>
+
 {# announcements and system messages #}
 {% if not activities.number > 1 %}
 <a href="{{ request.path }}" class="transition-y is-hidden notification is-primary is-block" data-poll-wrapper>
     {% blocktrans with tab_key=tab.key %}load <span data-poll="stream/{{ tab_key }}">0</span> unread status(es){% endblocktrans %}
+    {{ allowed_status_types|json_script:"unread-notifications-wrapper" }}
```

Comment:
Django doesn't have a cleaner way of inserting JSON values into the code. At least it's safely inserted.